sympy/solvers/tests/test_inequalities.py

```diff
@@ -323,16 +340,24 @@ def test_issue_10198():
         -1 + 1/abs(1/x - 1) < 0) == Or(
         And(-oo < x, x < 0), And(S(0) < x, x < S(1)/2)
         )
-    raises(NotImplementedError, lambda: reduce_inequalities(abs(1/sqrt(x)) - 1, x))
+    assert reduce_inequalities(abs(1/sqrt(x)) - 1, x) == Eq(x, 1)
     assert reduce_abs_inequality(-3 + 1/abs(1 - 1/x), '<', x) == \
         Or(And(-oo < x, x < 0),
         And(S(0) < x, x < S(3)/4), And(S(3)/2 < x, x < oo))
     raises(ValueError,lambda: reduce_abs_inequality(-3 + 1/abs(
         1 - 1/sqrt(x)), '<', x))
 
+
 def test_issue_10047():
-    assert solve(sin(x) < 2) == And(-oo < x, x < oo)
+    assert solve(sin(x) < 2) == And(S(0) <= x, x < 2*pi)
```

Comment:
> My issue is that this solution is wrong, unless you explicitly specify that the domain is [0, 2*pi).

Yes. I agree with you that should not be the solution unless the user specifies the domain to be
`[0, 2*pi)`. 
I think it will be better if we do this an XFAIL for now.
With the implementation of `BigUnion`, we would be able to un-XFAIL it later.

I have, however, documented this behaviour [here](https://github.com/sympy/sympy/pull/11458/files#diff-8c6b1223d1a80f79ca2509e14ebae02eR408). 
Shouldn't this suffice for now ?